AEPCampaignClassic/Sources/CampaignClassic.swift

```diff
@@ -46,27 +47,112 @@ public class CampaignClassic: NSObject, Extension {
+    ///
+    /// - Parameters:
+    ///   - event : the incoming track event
+    ///   - tagId : an integer string indicating whether it is a notification receive or notification click request
+    private func handleTrackEvent(event: Event, withTagId tagId: String) {
+        let configuration = CampaignClassicConfiguration.init(forEvent: event, runtime: runtime)
+
+        guard let trackingServer = configuration.trackingServer, !trackingServer.isEmpty else {
+            Log.debug(label: CampaignClassicConstants.LOG_TAG, "Unable to process TrackNotification request, Configuration not available.")
+            return
+        }
+
+        if configuration.privacyStatus != PrivacyStatus.optedIn {
+            Log.debug(label: CampaignClassicConstants.LOG_TAG, "Unable to process TrackNotification request, MobilePrivacyStatus is not optedIn.")
+            return
+        }
+
+        guard let deliveryId = event.deliveryId, !deliveryId.isEmpty else {
+            Log.debug(label: CampaignClassicConstants.LOG_TAG, "Unable to process TrackNotification request, trackingInfo deliveryId is nil (missing key `_dId` from tracking Info).")
+            return
+        }
+
+        guard let broadlogId = event.broadlogId, !broadlogId.isEmpty else {
+            Log.debug(label: CampaignClassicConstants.LOG_TAG, "Unable to process TrackNotification request, trackingInfo broadLogId is nil (missing key `_mId` from tracking Info).")
+            return
+        }
+
+        // V8 messageId is received in UUID format while V7 still comes as an integer(decimal) represented as a string.
+        // No transformation is required for the V8 UUID however for V7, message Id is parsed as an integer and converted to hex string.
+        guard let transformedBroadlogId = transformBroadLogId(broadlogId) else {
+            Log.debug(label: CampaignClassicConstants.LOG_TAG, "TrackingInfo broadLogId is nil (Missing key `_mId` from tracking Info), discarding the campaign classic track event.")
+            return
+        }
+
+        guard let trackingUrl = URL(string: String(format: CampaignClassicConstants.TRACKING_API_URL_BASE, trackingServer, transformedBroadlogId, deliveryId, tagId)) else {
+            return
+        }
+
+        let request = NetworkRequest(url: trackingUrl, httpMethod: .get, connectPayload: "", httpHeaders: [:], connectTimeout: configuration.timeout, readTimeout: configuration.timeout)
+
+        networkService.connectAsync(networkRequest: request, completionHandler: { connection in
+            if connection.responseCode == 200 {
+                Log.debug(label: CampaignClassicConstants.LOG_TAG, "TrackNotification success. URL : \(trackingUrl.absoluteString)")
+                return
+            }
+
+            Log.debug(label: CampaignClassicConstants.LOG_TAG, "Unable to trackNotification, Network Error. Response Code: \(String(describing: connection.responseCode)) URL : \(trackingUrl.absoluteString)")
```

Comment:
nit: the code isn't incorrect, but it feels backwards to check for the success path and return out of the completion block vs. checking failure paths and having the fall-through be the success condition 😛

Reply:
reading it back.. it feels weird to me as well

Changed back to the way how our brain is trained 😉